bndtools.core/src/bndtools/editor/project/AvailableBundlesPart.java

```diff
@@ -16,89 +14,65 @@
 import aQute.bnd.build.WorkspaceRepository;
 import aQute.bnd.build.model.BndEditModel;
 import aQute.bnd.osgi.Constants;
 import aQute.bnd.service.RepositoryPlugin;
 import aQute.bnd.service.ResolutionPhase;
-import bndtools.Plugin;
 import bndtools.central.Central;
 import bndtools.central.RepositoriesViewRefresher;
 import bndtools.central.RepositoryUtils;
 import bndtools.editor.common.BndEditorPart;
+import bndtools.jface.util.FilteredTree;
+import bndtools.jface.util.TreeFilter;
 import bndtools.model.repo.RepositoryBundle;
 import bndtools.model.repo.RepositoryTreeContentProvider;
 import bndtools.model.repo.RepositoryTreeLabelProvider;
 import bndtools.utils.SelectionDragAdapter;
 
 public class AvailableBundlesPart extends BndEditorPart implements RepositoriesViewRefresher.RefreshModel {
 
-	// Number of millis to wait for the user to stop typing in the filter box
-	private static final long					SEARCH_DELAY			= 1000;
-
-	private String								searchStr				= "";
-	private ScheduledFuture<?>					scheduledFilterUpdate	= null;
-
-	private final RepositoryTreeContentProvider	contentProvider			= new RepositoryTreeContentProvider(
+	private final RepositoryTreeContentProvider	contentProvider		= new RepositoryTreeContentProvider(
 		ResolutionPhase.runtime);
-	private Text								txtSearch;
 	private TreeViewer							viewer;
 
 	private Set<String>							includedRepos;
 
-	private final ViewerFilter					includedRepoFilter		= new ViewerFilter() {
-																			@Override
-																			public boolean select(Viewer viewer,
-																				Object parentElement, Object element) {
-																				boolean select = false;
-																				if (element instanceof RepositoryBundle) {
-																					RepositoryBundle repoBundle = (RepositoryBundle) element;
-																					RepositoryPlugin repo = repoBundle
-																						.getRepo();
-
-																					if (includedRepos == null) {
-																						select = true;
-																					} else if (repo instanceof WorkspaceRepository) {
-																						select = includedRepos
-																							.contains("Workspace");
-																					} else {
-																						select = includedRepos.contains(
-																							repoBundle.getRepo()
-																								.getName());
-																					}
-																				} else {
+	private final ViewerFilter					includedRepoFilter	= new ViewerFilter() {
+																		@Override
+																		public boolean select(Viewer viewer,
+																			Object parentElement, Object element) {
+																			boolean select = false;
+																			if (element instanceof RepositoryBundle) {
+																				RepositoryBundle repoBundle = (RepositoryBundle) element;
+																				RepositoryPlugin repo = repoBundle
+																					.getRepo();
+
+																				if (includedRepos == null) {
 																					select = true;
```

Comment:
Sorry to be painful but it's difficult to review when there's these unrelated whitespace-only changes in the commit. 😦

Reply:
(Thanks for taking it on!)

Reply:
I've no idea how it can happen. We automatically format on save and the formatting instructions have not changed. Somehow Eclipse has some randomness in it. I agree it is a pain.

Reply:
Well, look what I found: 
![Github diff](https://github.com/bndtools/bnd/assets/7224087/066d94f4-489e-476f-874a-ee7e92157c9c)

I'll get to reviewing it then!